codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/restxml/httpResponse/AWSXMLHttpResponseTraitWithoutPayload.kt

```diff
@@ -31,62 +21,11 @@ class AWSXMLHttpResponseTraitWithoutPayload(
+        val bodyMembersWithoutQueryTrait = responseBindings.filter { it.location == HttpBinding.Location.DOCUMENT }
             .filter { !it.member.hasTrait(HttpQueryTrait::class.java) }
-            .map { ctx.symbolProvider.toMemberName(it.member) }
-            .toMutableSet()
-
-        if (bodyMembersWithoutQueryTrait.isNotEmpty()) {
-            writer.write("if let data = try await httpResponse.body.readData(), let responseDecoder = decoder {")
-            writer.indent()
-            val outputShapeName = ctx.symbolProvider.toSymbol(outputShape).name
-            if (serviceDisablesWrappingOfErrorProperties()) {
-                renderWithoutErrorResponseContainer(outputShapeName, bodyMembersWithoutQueryTrait)
-            } else {
-                renderWithErrorResponseContainer(outputShapeName, bodyMembersWithoutQueryTrait)
-            }
-            writer.dedent()
-            writer.write("} else {")
-            writer.indent()
-            bodyMembers.sortedBy { it.memberName }.forEach {
-                val memberName = ctx.symbolProvider.toMemberName(it.member)
-                val type = ctx.model.expectShape(it.member.target)
-                val value = if (ctx.symbolProvider.toSymbol(it.member).isBoxed()) "nil" else {
-                    when (type) {
-                        is IntegerShape, is ByteShape, is ShortShape, is LongShape -> 0
-                        is FloatShape, is DoubleShape -> 0.0
-                        is BooleanShape -> false
-                        else -> "nil"
-                    }
-                }
-                writer.write("self.properties.$memberName = $value")
-            }
-            writer.dedent()
-            writer.write("}")
-        }
-    }
-
-    fun serviceDisablesWrappingOfErrorProperties(): Boolean {
-        ctx.service.getTrait<RestXmlTrait>()?.let {
-            return it.isNoErrorWrapping
-        }
-        return false
-    }
-
-    fun renderWithoutErrorResponseContainer(outputShapeName: String, bodyMembersWithoutQueryTrait: Set<String>) {
-        writer.write("let output: ${outputShapeName}Body = try responseDecoder.decode(responseBody: data)")
-        bodyMembersWithoutQueryTrait.sorted().forEach {
-            writer.write("self.properties.$it = output.$it")
-        }
-    }
-
-    fun renderWithErrorResponseContainer(outputShapeName: String, bodyMembersWithoutQueryTrait: Set<String>) {
-        writer.addImport(ErrorResponseContainer)
-        writer.write("let output: \$N<${outputShapeName}Body> = try responseDecoder.decode(responseBody: data)", ErrorResponseContainer)
-        bodyMembersWithoutQueryTrait.sorted().forEach {
-            writer.write("self.properties.$it = output.error.$it")
-        }
+            .map { it.member }
+            .toSet()
+        val generator = MemberShapeDecodeXMLGenerator(ctx, writer, outputShape)
+        bodyMembersWithoutQueryTrait.sortedBy { it.memberName }.forEach { generator.render(it) }
```

Comment:
Again, pretty much this entire type's code generation is delegated to `MemberShapeDecodeXMLGenerator`.